module/zfs/arc.c

```diff
@@ -8218,6 +8218,11 @@ l2arc_write_size(l2arc_dev_t *dev)
 
 		size = l2arc_write_max = l2arc_write_boost = L2ARC_WRITE_SIZE;
 
+		if (l2arc_trim_ahead > 1) {
+			cmn_err(CE_NOTE, "l2arc_trim_ahead set to 1");
+			l2arc_trim_ahead = 1;
+		}
```

Comment:
I'd expect a common value here to be 100, why reset this to 1 when it's `>1`?

Reply:
I think it makes sense to make it the smallest possible, but still not disable it if it was enabled. This code applies only when the write size is exceeds the size of the device, for example in the case of a misconfiguration of those variables.

Reply:
Got it, that reasonable.